indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -96,6 +97,9 @@ public static IngestionMode fromString(String name)
   private final Map<String, Object> context;
   private File reportsFile;
 
+  @JsonIgnore
+  public static Map<String, Object> taskMetadata;
```

Comment:
This is assuming there is only one task per peon... That's a big assumption

Reply:
Would prefer to move this somewhere else or wire it down to where the exception is created via constructors.

Reply:
Any reason to not re-use the existing `context`?

Reply:
Any reason to not use the existing `context` map rather than adding another map?

Reply:
Might be good to allow the user to specify other metadata that they want to associate with the task.

Reply:
> This is assuming there is only one task per peon... That's a big assumption

This would be true for middlemanagers + peons (one task per process), but not true for indexers (multiple tasks run in same process)